server/historian/packages/historian-base/src/runnerFactory.ts

```diff
@@ -47,7 +49,8 @@ export class HistorianResourcesFactory implements utils.IResourcesFactory<Histor
         const tenantCache = new historianServices.RedisTenantCache(redisClient);
         // Create services
         const riddlerEndpoint = config.get("riddler");
-        const riddler = new historianServices.RiddlerService(riddlerEndpoint, tenantCache);
+        const asyncLocalStorage = config.get("asyncLocalStorageInstance");
```

Comment:
In order to retrieve a complex object through nconf.get, it has to be within an Array. 

At some point I looked into the nconf source code to see why, but I can't remember at the moment or never figured out why.

Here's some code I used to prove it at the time:

```js
const nconf = require('nconf');

class Logger {
    logData(data) {
        console.log(data);
    }
}

class FancyLogger extends Logger {
    logData(data) {
        console.log(data, { fancy: true });
    }
}

class Lumberjack {
    logger = new Logger();
    init(logger) {
        if (!logger instanceof Logger) {
            throw new Error("logger is not a Logger");
        }
        this.logger = logger;
    }
    log(data) {
        this.logger.logData(data);
    }
}

const config = nconf.use("memory");

// This implementation works
const lumberjack = new Lumberjack();
config.set("logger:loggerType", [new FancyLogger()]);
lumberjack.init(config.get("logger:loggerType")[0]);
lumberjack.log("Array Lumberjack");
/* Output:
Array Lumberjack { fancy: true }
*/

// This implementation breaks
const brokenLumberjack = new Lumberjack();
config.set("logger:loggerType", new FancyLogger());
brokenLumberjack.init(config.get("logger:loggerType"));
brokenLumberjack.log("Broken Lumberjack");
/* Output:
this.logger.logData(data);
                    ^

TypeError: this.logger.logData is not a function
    at Lumberjack.log (/Users/znewton/Code/tmp-tests/configShenanigans.js:24:21)
    at Object.<anonymous> (/Users/znewton/Code/tmp-tests/configShenanigans.js:63:18)
    at Module._compile (internal/modules/cjs/loader.js:999:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
    at Module.load (internal/modules/cjs/loader.js:863:32)
    at Function.Module._load (internal/modules/cjs/loader.js:708:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:60:12)
    at internal/main/run_main_module.js:17:47
*/
```

Reply:
You are right, missed that.. Added in the latest update.